cli/azd/cmd/root.go

```diff
@@ -93,6 +93,7 @@ For more information, visit the Azure Developer CLI Dev Hub: https://aka.ms/azur
 	cmd.AddCommand(upCmd(opts))
 	cmd.AddCommand(templatesCmd(opts))
 	cmd.AddCommand(versionCmd(opts))
+	cmd.AddCommand(telemetryCmd(opts))
```

Comment:
What do we think about adding the telemetry as a `cmd` instead of a special flag?

Personally, I lean towards a command since it avoids adding additional "special paths", and we have less branching at the `main` entry point, rather we treat each command as the sub entry point.

It also opens up possibilities for future extension. For example, `azd telemetry list` to show current to-be-uploaded telemetry or `azd telemetry cleanup`.

Reply:
I'm also using `azd telemetry upload --debug` as a way to enable logging for the telemetry upload.